misc/python/materialize/checks/all_checks/upsert.py

```diff
@@ -179,3 +179,59 @@ def validate(self) -> Testdrive:
            """
             )
         )
+
+
+class UpsertLegacy(Check):
+    """
+    An upsert source test that uses the legacy syntax to create the source
+    on all versions to ensure the source is properly migrated with the
+    ActivateSourceVersioningMigration scenario
+    """
+
+    def initialize(self) -> Testdrive:
+        return Testdrive(
+            schemas()
+            + dedent(
+                """
+                $ kafka-create-topic topic=upsert-legacy-syntax
```

Comment:
I guess we should explicitly disable `force_source_table_syntax` here.

Reply:
It won't really matter to do that here - the flag `force_source_table_syntax` currently only matters when Materialize is booting during the migration phase, so it will only activate the migration if it's applied as a `system_parameter_default` or in LD.

In the `ActivateSourceVersioningMigration` we activate the flag on the second instance of `StartMz` which occurs after this `initialize()` method is run, such that these statements are then migrated to the new statements before `validate()` is run.